### PR TITLE
Added a configuration option to disable auto dependency updates

### LINKED ISSUE #2235: Add a configuration option to disable auto dependency updates
We recently added a feature where Python dependencies are automatically updated when launching GDB. In the majority of cases this is best to leave on by default, but we should add a configuration option in case anyone wants to disable this functionality in some specific cases.

The thing is that pwndbg's normal configuration infra isn't loaded yet, so we can't use it in `gdbinit.py`. Maybe we just use an environment variable?

### PR DESCRIPTION
**Pull request by B1N4RY-P4R45173**
Closes #2235

### CODE REVIEW

**Comment by B1N4RY-P4R45173** [comment]
Okay, it took me some time to realize why some of these tests were failing earlier. There were primarily two reasons:

1) The GitHub tests raise errors when interactive commands such as `read` are passed even if we pass a default answer with timeout.

2) Setting up permanent env variables also seems to be a tricky task.

I was trying to set a permanent env variable to store the user's choice about auto updates, so we can check it from `gdbinit.py` and run appropriate command if update is needed.

One alternative could be to add a file marking the user's choice only if he chooses to turn off the auto update of dependencies. And when `gdbinit.py` is run, if the file (the one used for marking) exists, we can skip or else we need to update the dependencies

**Comment by B1N4RY-P4R45173** [comment]
I think we might also need slight changes in the workflow files of these tests and the docker files

**Comment by B1N4RY-P4R45173** [comment]
@disconnect3d I need some help with this

File: gdbinit.py [hunk 1]
```diff
@@ -168,7 +168,12 @@ def main() -> None:
             sys.stdout.flush()
             os._exit(1)
 
-        update_deps(src_root, venv_path)
+        # Check for the environment variable to disable auto-update
+        auto_update_disabled = os.getenv("PWNDBG_AUTO_UPDATE", "true").lower() == "false"
```
Comment (disconnect3d) [inline gdbinit.py]: Let's change this to `PWNDBG_NO_AUTOUPDATE` and if it is defined with any value, skip updating deps.

File: setup.sh [hunk 2]
```diff
@@ -200,8 +200,59 @@ ${PYTHON} -m venv -- ${PWNDBG_VENV_PATH}
...
+            ;;
+        *)
+            echo "Unsupported shell. You will need to manually add your shell config file."
+            return 1
+            ;;
+    esac
+}
+
+# Function to persist the environment variable in the appropriate shell configuration file
+persist_env_var() {
```
Comment (disconnect3d) [inline setup.sh]: We dont't want any of this. Let's just inform user on setup they can disable auto update checks with a given envvar and let the user to add it or not by themselves.

Comment (disconnect3d) [reply setup.sh]: I.e. we don't want to handle sh/bash/zsh/fish/bazilion other shells; the users can do it on their own

**Review by disconnect3d — CHANGES_REQUESTED** [review]
As discussed on Discord, lets simplify this

File: gdbinit.py (hunk 1 above)
Comment (B1N4RY-P4R45173) [reply gdbinit.py]: Okay I will change it asap.

File: setup.sh (hunk 2 above)
Comment (B1N4RY-P4R45173) [reply setup.sh]: Sure I got it. Will complete it by today.

**Review by disconnect3d — APPROVED** [review]

**Comment by disconnect3d** [comment]
Thanks!

**Pull request merged**